app/dashboard/e2e/actions/BaseActions.ts

```diff
@@ -155,4 +155,24 @@ export default class BaseActions implements Promise<void> {
       await page.keyboard.up(await actions.modModifier(page))
     })
   }
+
+  /** Expect an input to have an error (or no error if the expected value is `null`).
+   * If the expected value is `undefined`, the assertion is skipped. */
+  expectInputError(name: string, description: string, expected: string | null | undefined) {
+    if (expected === undefined) {
+      return this
+    } else if (expected != null) {
+      return this.step(`Expect ${description} error to be '${expected}'`, async (page) => {
+        // eslint-disable-next-line no-restricted-properties
+        const element = page.locator(`fieldset:has(input[name=${name}]) > span.text-danger`)
```

Comment:
This is a fragile selector. Can we use `testid` or something like that here?

Reply:
the entire thing? or the `.text-danger` specifically?
i'm thinking of keeping the `fieldset:has()` since i don't think *that* part is too fragile, but do agree that the `text-danger` isn't great

Reply:
Both 'fieldset', '>' and 'text-danger' are fragile parts here

Reply:
adjusted, but not sure whether the current solution is ideal. the label containing the input is causing a bit of awkwardness (we can't use exact locators)

Reply:
WDYM? Why we can't use something like: 

```ts
const error = page.getByTestId('password-input').getByTestId('error-text')
```

Reply:
i guess the idea is that test-id should be a last resort, but i guess it might be our best option here?

Reply:
Yeah, I don't see better options either